### PR TITLE
Cividesk getting started 4.6

### PR DESCRIPTION
**Pull request by colemanw**

### CODE REVIEW

**Comment by colemanw** [comment]
@totten any idea why the style checker is going berserk on this PR?

File: CRM/Upgrade/Incremental/php/FourSix.php [hunk 1]
```diff
@@ -258,4 +258,38 @@ public static function task_4_6_x_runOnlySql(CRM_Queue_TaskContext $ctx, $rev) {
     return TRUE;
   }
 
+
+  /**
+   * Upgrade function.
+   *
+   * @param string $rev
+   */
+  public function upgrade_4_6_7($rev) {
```
Comment (colemanw) [inline CRM/Upgrade/Incremental/php/FourSix.php]: this will need to be updated before merging

**Comment by colemanw** [comment]
@sunilpawar check the [test results](https://test.civicrm.org/job/CiviCRM-Core-PR/6652/) - the upgrade code is crashing.

File: CRM/Upgrade/Incremental/php/FourSix.php (hunk 1 above)
Comment (sunilpawar) [inline CRM/Upgrade/Incremental/php/FourSix.php]: what change required here, latest version ?

Comment (colemanw) [inline CRM/Upgrade/Incremental/php/FourSix.php]: latest + 1 = 4.6.11

**Comment by eileenmcnaughton** [comment]
@nganivet What is the deal with this - I feel like we have Getting Started in 4.6 already but maybe I imagine it? This needs a minor fix to the upgrade script - but I'm not clear if it is outdated

**Comment by nganivet** [comment]
@eileenmcnaughton We submitted our PR on July 31, but it was not merged until Nov 23. In the meantime new releases had been published, so the upgrade code needs to be moved to the function corresponding to the latest published release.

@colemanw I would guess the test fail because the test database already had been upgraded to include the widget (the tests passed when we had initially submitted out PR, and there is no code change). Can you please take care of this since it took you so long to merge?

**Comment by eileenmcnaughton** [comment]
Since the code in the PR does not add any new tests :-( the fails were probably unrelated to the upgrade issue. I'll kick them off again & we can see why

In general it's probably better to try to add the main part of the functionality and tests for any new behaviour in a separate PR to the upgrade script as it's particularly hard to stop that from going stale.

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by colemanw** [comment]
jenkins, test this please

**Comment by eileenmcnaughton** [comment]
Since I've spent most of today doing other people's dishes I did the rebase (and squash) on this and made the required one line change that blocked this from being merged.

I'm not sure about the failing test at the moment - but if it doesn't pass on my revised PR I'm inclined to close it on the basis that if this only goes into 4.7 I think that's fine. This PR has been open for 5 months and it doesn't look like anyone has been shepherding it through from CiviDesk or core - which I think is proxy for 'getting this in 4.6 is not vital'

https://github.com/civicrm/civicrm-core/pull/7559

From recollection this was not merged into 4.4 for the same reason - ie. it was submitted too late for the release at which the upgrade script was targetted at.  I guess both CiviDesk and core could reasonably make the case that it was not their responsibility - but one thing that worries me about this situation is that having PRs sit open blows out the PR queue to the point where it's easy to miss PRs coming from new contributors and hard to keep on top of.

**Comment by nganivet** [comment]
Eileen - did not really know we had to 'shepherd' PR's once submitted to 
core. Your comment also seem to imply that the PR queue is getting 
stale. This all begs for some clarification as to what is the 'PR queue 
process' and core/partners role in this. Has this been though 
of/documented before?

------ Original Message ------

**Comment by eileenmcnaughton** [comment]
I have been looking at the queue a lot over the last couple of days and there are definitely a number of stale PRs for various reasons. There are also some in the 'this is never actually going to be finished category' which are clogging up the works.

I think where thinks are falling down the gaps a bit are that the core team/ community reviewers (she says optimistically)  are generally waiting until a PR passes the tests before reviewing rather than giving feedback that says 'I would be OK to merge this provided x is fixed, unit tests are added for the new function and the existing style and test errors are fixed'. This is hard, work intensive feedback to give - often involving more work than the original code fix so it's understandable to wait until the code is 'finished' - ie. tests are passing. In addition to showing the code has a certain quality it also indicates the contributor has a commitment to finishing the job.

However, often there is a big lag in the tests being brought to the passing stage, by which time the PR is no longer at the top of the list so it could be a while before someone notices it is now ready to look at. By which time it's really hard to insist on tests being added and often the PR is stale  putting it back into the cycle and dragging it out for everyone.

One thing we have started is adding a 'merge on pass' label to the PR, which helps. One of the limits of labels is they only can be edited by certain people. Yesterday Tim & I discussed putting some pseudo labels in the PR title - which a  comment explaining. Eg. if a PR is stale prepending a title with [needs-rebase] and adding a comment to explain to remove that later. We could do that with other titles like 

[extra-work-requested]
[needs-test]

and [test-incl]

- the latter being equivalent to -priotise assisting on this one - although it might be better as a label?

**Pull request closed without merging**